### PR TITLE
[WIP] Issue #2676 enable validating and mutation admission webhook

### LINKED ISSUE #2676: Enable mutating and validating webhooks by default
Mutating and validating webhooks are central to the story of extending Kubernetes. We should ensure the experience of evaluating or running Kubernetes extensions that use these features is as painless as possible in minishift.

When getting [Knative](https://github.com/knative) to run with OpenShift, we found this to be the formula to use to get webhooks enabled:

https://github.com/knative/docs/blob/master/install/Knative-with-OpenShift.md#creating-a-new-openshift-cluster

I'd like to re-write the linked instructions to use minishift instead of oc cluster up; in order to do that we need validating and mutating webhooks to be enabled by default.  See also: https://github.com/openshift/origin/issues/20530

### PR DESCRIPTION
**Pull request by anjannath**
Fixes #2676

### CODE REVIEW

**Comment by anjannath** [comment]
retest this please

**Comment by anjannath** [comment]
CI failure is due to changes in the post cluster up need to look into this.

**Comment by anjannath** [comment]
retest this please

File: cmd/minishift/cmd/openshift/set.go [hunk 1]
```diff
@@ -78,6 +78,8 @@ func runPatch(cmd *cobra.Command, args []string) {
 	sshCommander := provision.GenericSSHCommander{Driver: host.Driver}
 	dockerCommander := docker.NewVmDockerCommander(sshCommander)
 
+	fmt.Println(fmt.Sprintf("Patching OpenShift configuration '%s' with '%s'", patchTarget.LocalConfigFile, patch))
```
Comment (LalatenduMohanty) [inline cmd/minishift/cmd/openshift/set.go]: This should be behind Verbose 2.

**Comment by LalatenduMohanty** [comment]
@anjannath Is there anything pending from the code point of view?

File: cmd/minishift/cmd/openshift/set.go (hunk 1 above)
Comment (LalatenduMohanty) [reply cmd/minishift/cmd/openshift/set.go]: Also I am not sure why this Println is present here. Was it for debugging?

**Comment by anjannath** [comment]
@LalatenduMohanty The patch is applied, i need to add code to check if the container is up and then pass on control.

File: cmd/minishift/cmd/openshift/set.go (hunk 1 above)
Comment (anjannath) [reply cmd/minishift/cmd/openshift/set.go]: Yes, this was for debugging.

**Comment by anjannath** [comment]
retest this please

File: cmd/minishift/cmd/openshift/set.go (hunk 1 above)
Comment (anjannath) [reply cmd/minishift/cmd/openshift/set.go]: Nope, this is not for debugging, this println was present in the `pkg/openshift/openshift.go` i moved it to the cmd package, its a refactor

**Comment by anjannath** [comment]
@LalatenduMohanty I am  thinking of closing this in favour of #3044.

There's no way to correctly know if the cluster is up and able to serve, after patching the kube config file, as a result all the features in the integration test that do a `minishift start` and some commands after that needs to be changed to have some wait period/retry in them.

The only way to correctly know if the cluster is up is to run some `oc` commands against it. Adding the waits in post cluster up introduces delays in `minishift start`, which is bad IMO.

/cc @praveenkumar

**Comment by LalatenduMohanty** [comment]
> @LalatenduMohanty I am thinking of closing this in favour of #3044.

Works for me. I am closing this cc @praveenkumar

**Pull request closed without merging**